src/main.c

```diff
@@ -46,6 +47,24 @@ static void user_configuration(bool run)
 	set_speed_mode(mode, run);
 }
 
+/**
+ * @brief Check battery voltage and warn if the voltage is getting too low.
+ */
+static void check_battery_voltage(void)
+{
+	float voltage;
+
+	voltage = get_battery_voltage();
+	if (voltage < 3.6)
+		speaker_warn_low_battery();
+	if (voltage < 3.5)
+		speaker_warn_low_battery();
+	if (voltage < 3.4)
+		speaker_warn_low_battery();
+	if (voltage < 3.3)
+		speaker_play('C', 3, 0, 2.);
```

Comment:
I would create a function on hmi for this case, so speaker is only connected to hmi and is clearer